### PR TITLE
Add contentDescription field to fileTypeSummaries (#3170)

### PR DESCRIPTION
**Pull request by danielsotirhos**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab                               <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in sandbox and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [ ] Started reindex in `dev`                              <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [ ] Checked for failures in `dev`                         <sub>or this PR does not require reindexing or does not target `dev`</sub>
- [ ] Started reindex in `prod`                             <sub>or this PR does not require reindexing or does not target `prod`</sub>
- [ ] Checked for failures in `prod`                        <sub>or this PR does not require reindexing or does not target `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

**Review by achave11 — APPROVED** [review]
LGTM

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Now the big question is if DB wants to group the the fileTypeSummary by content description or not. Could you ask them on Slack and ping me?

**Review by hannes-ucsc — APPROVED** [review]

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
This changes the API semantics and is potentially backwards incompatible. Could you ask the DB team how they prefer us to handle the deployment? Last time I requested to be pinged on the conversation, please make sure you do so this time. No bump.

**Comment by danielsotirhos** [comment]
> Could you ask the DB team how they prefer us to handle the deployment? 

https://ucsc-gi.slack.com/archives/C705Y6G9Z/p1625096980236300?thread_ts=1625096958.236100&cid=C705Y6G9Z

File: test/indexer/data/fa5be5eb-2d64-49f5-8ed8-bd627ac9bc7a.metadata.json [hunk 1]
```diff
@@ -393,7 +407,14 @@
         "file_core": {
             "file_name": "pbmc4k_1000_S1_L001_I1_001.fastq.gz",
             "file_format": "fastq.gz",
-            "checksum": "3d252f83492bf030ddfd05a5a1411cdc"
+            "checksum": "3d252f83492bf030ddfd05a5a1411cdc",
+            "content_description": [
```
Comment (hannes-ucsc) [inline test/indexer/data/fa5be5eb-2d64-49f5-8ed8-bd627ac9bc7a.metadata.json:411]: The problem is that the [declared schema](http://schema.staging.data.humancellatlas.org/type/file/7.0.2/sequence_file) does not allow this. Can we cover this differently? We should also cover the grouping aspect. There are a few canned bundles that already have `content_description` so maybe we need to bite the bullet and add a test specifically for `fileTypeSummaries`.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**